src/pybind/mgr/rook/module.py

```diff
@@ -442,10 +442,17 @@ def apply_mon(self, spec):
 
         return self.rook_cluster.update_mon_count(spec.placement.count)
 
+    def get_num_osds(self) -> int:
+        num_of_osds = self.get_ceph_option('osd_pool_default_size')
+        if type(num_of_osds) is not int:
+            raise RuntimeError("osd_pool_default_size ceph option is not integer")
+        return num_of_osds
+
     @handle_orch_error
     def apply_mds(self, spec):
         # type: (ServiceSpec) -> str
-        return self.rook_cluster.apply_filesystem(spec)
+        num_of_osds = self.get_num_osds()
```

Comment:
drop the get_num_osds() function and just call self.get_ceph_option here.  it will always be an integer, so no need for the error check.